instrumentation/jmx-metrics/javaagent/src/main/java/io/opentelemetry/instrumentation/javaagent/jmx/JmxMetricInsightInstaller.java

```diff
@@ -0,0 +1,102 @@
+import io.opentelemetry.instrumentation.jmx.engine.MetricConfiguration;
+import io.opentelemetry.instrumentation.jmx.yaml.RuleParser;
+import io.opentelemetry.javaagent.extension.AgentListener;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import java.io.File;
+import java.io.InputStream;
+import java.nio.file.Files;
+
+/** An {@link AgentListener} that enables JMX metrics during agent startup. */
+@AutoService(AgentListener.class)
+public class JmxMetricInsightInstaller implements AgentListener {
+
+  @Override
+  public void afterAgent(AutoConfiguredOpenTelemetrySdk autoConfiguredSdk) {
+    ConfigProperties config = autoConfiguredSdk.getConfig();
+
+    if (config.getBoolean("otel.jmx.enabled", true)) {
+      JmxMetricInsight service =
+          JmxMetricInsight.createService(GlobalOpenTelemetry.get(), beanDiscoveryDelay(config));
+      MetricConfiguration conf = buildMetricConfiguration();
+      service.start(conf);
+    }
+  }
+
+  private static long beanDiscoveryDelay(ConfigProperties configProperties) {
+    Long discoveryDelay = configProperties.getLong("otel.jmx.discovery.delay");
+    if (discoveryDelay != null) {
+      return discoveryDelay;
+    }
+
+    // If discovery delay has not been configured, have a peek at the metric export interval.
+    // It makes sense for both of these values to be similar.
+    long exportInterval = configProperties.getLong("otel.metric.export.interval", 60000);
+    return exportInterval;
+  }
+
+  private static String resourceFor(String platform) {
+    return "/jmx/rules/" + platform + ".yaml";
+  }
+
+  private static void addRulesForPlatform(String platform, MetricConfiguration conf) {
+    String yamlResource = resourceFor(platform);
+    try (InputStream inputStream =
+        JmxMetricInsightInstaller.class.getResourceAsStream(yamlResource)) {
+      if (inputStream != null) {
+        JmxMetricInsight.getLogger().log(FINE, "Opened input stream {0}", yamlResource);
+        RuleParser parserInstance = RuleParser.get();
+        parserInstance.addMetricDefs(conf, inputStream);
+      } else {
+        JmxMetricInsight.getLogger().log(CONFIG, "No support found for {0}", platform);
+      }
+    } catch (Exception e) {
+      JmxMetricInsight.getLogger().warning(e.getMessage());
+    }
+  }
+
+  private static void buildFromDefaultRules(MetricConfiguration conf) {
+    String targetSystem = System.getProperty("otel.jmx.target.system", "").trim();
+    String[] platforms = targetSystem.length() == 0 ? new String[0] : targetSystem.split(",");
+
+    for (String platform : platforms) {
+      addRulesForPlatform(platform, conf);
+    }
+  }
+
+  private static void buildFromUserRules(MetricConfiguration conf) {
+    String jmxDir = System.getProperty("otel.jmx.config");
+    if (jmxDir != null) {
+      JmxMetricInsight.getLogger().log(CONFIG, "JMX config file name: {0}", jmxDir);
```

Comment:
I don't think we use CONFIG level in other places, it took me a bit to track down how that would map to slf4j which is the eventual destination in the javaagent ([looks like it is mapped to INFO](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/a8e590024489d0a5ddf86c0ff2159b9c4192b13b/javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/PatchLogger.java#L184-L196))

since the destination isn't really jul, i'd suggest using `FINE` here

Reply:
Yes, I'll change that.